src/zope/sendmail/tests/test_mailer.py

```diff
@@ -128,20 +134,21 @@ def test_send_multiple_same_mailer(self):
         # multiple mails.
         smtps = []
 
-        def hook(smtp):
-            smtps.append(smtp)
+        # Note: this test needs to be thread-safe so it must avoid mutating
+        # attributes on `self`!
 
-        self.smtp_hook = hook
         for run in (1, 2):
```

Comment:
The line `mailer = self._makeMailer(smtp_hook=smtps.append)` should be called before starting the `for` loop.